File: ObjectiveC/FlexGrid101/FlexGrid101/EditConfirmationController.m
//
//  EditConfirmationController.m
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "EditConfirmationController.h"
#import "CustomerData.h"
#import "XuniFlexGridDynamicKit/XuniFlexGridDynamicKit.h"

@interface EditConfirmationController ()
@property (weak, nonatomic) IBOutlet FlexGrid *flex;
@end

@implementation EditConfirmationController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.flex.isReadOnly = false;
    
    self.flex.columnHeaderFont = [UIFont boldSystemFontOfSize:self.flex.columnHeaderFont.pointSize];
    self.flex.autoGenerateColumns = false;
    
    GridColumn *identifier = [[GridColumn alloc] init];
    identifier.binding = @"customerID";
    identifier.isReadOnly = true;
    identifier.widthType = GridColumnWidthPixel;
    identifier.width = 100;
    [self.flex.columns addObject:identifier];
    
    GridColumn *firstName = [[GridColumn alloc] init];
    firstName.binding = @"firstName";
    [self.flex.columns addObject:firstName];
    
    GridColumn *lastName = [[GridColumn alloc] init];
    lastName.binding = @"lastName";
    [self.flex.columns addObject:lastName];
    
    GridColumn *address = [[GridColumn alloc] init];
    address.binding = @"address";
    [self.flex.columns addObject:address];
    
    GridColumn *city = [[GridColumn alloc] init];
    city.binding = @"city";
    [self.flex.columns addObject:city];
    
    GridColumn *postalCode = [[GridColumn alloc] init];
    postalCode.binding = @"postalCode";
    [self.flex.columns addObject:postalCode];
    
    GridColumn *active = [[GridColumn alloc] init];
    active.binding = @"active";
    active.widthType = GridColumnWidthPixel;
    active.width = 70;
    [self.flex.columns addObject:active];
    
    self.flex.itemsSource = [CustomerData getCustomerData:100];
    
    [self.flex.flexGridCellEditEnding addHandler:^(XuniEventContainer<GridCellRangeEventArgs *> *eventContainer) {
        NSObject* activeEditorValue = self.flex.activeEditorValue;
        
        if(activeEditorValue == nil) activeEditorValue = @"";
        
        if([[self.flex.cells getCellDataForRow:eventContainer.eventArgs.row inColumn:eventContainer.eventArgs.col formatted:false] isEqual:activeEditorValue])
        {
            eventContainer.eventArgs.cancel = true;
            return;
        }
        
        NSObject* prev = [self.flex.cells getCellDataForRow:eventContainer.eventArgs.row inColumn:eventContainer.eventArgs.col formatted:false];
        
        NSString *title = [[NSString alloc] init];
        NSString *message = [[NSString alloc] init];
        title = NSLocalizedString(@"Edit Confirmation", nil);
        message = NSLocalizedString(@"Do you want to commit the edit?", nil);
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Ok", nil) style:UIAlertActionStyleDefault handler:^(UIAlertAction *action){
        }];
        UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", nil) style:UIAlertActionStyleCancel handler:^(UIAlertAction *action){
            [self.flex.cells setCellData:prev forRow:eventContainer.eventArgs.row inColumn:eventContainer.eventArgs.col];
        }];
        
        
        [alertController addAction:okAction];
        [alertController addAction:cancelAction];
        [self presentViewController:alertController animated:true completion:nil];
        
        eventContainer.eventArgs.cancel = false;

    } forObject:self];
}


@end

File: Swift/FlexGrid101/FlexGrid101/EditConfirmationController.swift
//
//  EditConfirmationController.h
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//
import UIKit
import XuniFlexGridDynamicKit

class EditConfirmationController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.flex.isReadOnly = false
        self.flex.columnHeaderFont = UIFont.boldSystemFontOfSize(self.flex.columnHeaderFont.pointSize)
        self.flex.autoGenerateColumns = false
        let identifier: GridColumn = GridColumn()
        identifier.binding = "customerID"
        identifier.isReadOnly = true
        identifier.widthType = GridColumnWidth.Pixel
        identifier.width = 100
        self.flex.columns.addObject(identifier)
        let firstName: GridColumn = GridColumn()
        firstName.binding = "firstName"
        self.flex.columns.addObject(firstName)
        let lastName: GridColumn = GridColumn()
        lastName.binding = "lastName"
        self.flex.columns.addObject(lastName)
        let address: GridColumn = GridColumn()
        address.binding = "address"
        self.flex.columns.addObject(address)
        let city: GridColumn = GridColumn()
        city.binding = "city"
        self.flex.columns.addObject(city)
        let postalCode: GridColumn = GridColumn()
        postalCode.binding = "postalCode"
        self.flex.columns.addObject(postalCode)
        let active: GridColumn = GridColumn()
        active.binding = "active"
        active.widthType = GridColumnWidth.Pixel
        active.width = 70
        self.flex.columns.addObject(active)
        
        
        self.flex.itemsSource = NSMutableArray(array: CustomerData.getCustomerData(100))
        
        self.flex.flexGridCellEditEnding.addHandler({(eventContainer: XuniEventContainer!) -> Void in
            let eventContainerEventArgs : GridCellRangeEventArgs = eventContainer.eventArgs as! GridCellRangeEventArgs
            
            var activeEditorValue: NSObject! = self.flex.activeEditorValue
            if activeEditorValue == nil {
                activeEditorValue = ""
            }
            if self.flex.cells.getCellDataForRow(eventContainerEventArgs.row, inColumn: eventContainerEventArgs.col, formatted: false).isEqual(activeEditorValue) {
                eventContainerEventArgs.cancel = true
                return
            }
            let prev: NSObject = self.flex.cells.getCellDataForRow(eventContainerEventArgs.row, inColumn: eventContainerEventArgs.col, formatted: false)
            var title: String = String()
            var message: String = String()
            title = NSLocalizedString("Edit Confirmation", comment: "")
            message = NSLocalizedString("Do you want to commit the edit?", comment: "")
            let alertController: UIAlertController = UIAlertController(title: title, message: message, preferredStyle: .Alert)
            let okAction: UIAlertAction = UIAlertAction(title: NSLocalizedString("Ok", comment: ""), style: .Default, handler: {(action: UIAlertAction) -> Void in
                })
            let cancelAction: UIAlertAction = UIAlertAction(title: NSLocalizedString("Cancel", comment: ""), style: .Cancel, handler: {(action: UIAlertAction) -> Void in
                    self.flex.cells.setCellData(prev, forRow: eventContainerEventArgs.row, inColumn: eventContainerEventArgs.col)
                })
            alertController.addAction(okAction)
            alertController.addAction(cancelAction)
            self.presentViewController(alertController, animated: true, completion: { _ in })
            eventContainerEventArgs.cancel = false
        }, forObject: self)
    }

    @IBOutlet weak var flex: FlexGrid!
}
//
//  EditConfirmationController.m
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//